### PR TITLE
tests: Test edge cases for VBO robustness with weird strides

### PR DESCRIPTION
**Pull request by HansKristian-Work**
- Passes WARP.
- Expected failures on AMD native windows driver are marked. The range check behaves as expected if we were to ignore dynamic stride rules in Vulkan. Elements which should be OOB are not marked as such.
- Fails quite a lot of RADV. Will file bugs.
- (Need to test NV)

### CODE REVIEW

File: libs/vkd3d/command.c [hunk 1]
```diff
@@ -4764,27 +4764,32 @@ static void d3d12_command_list_update_dynamic_state(struct d3d12_command_list *l
...
+                    dyn_state->viewport_count, dyn_state->scissors));
+        }
+        else if (dyn_state->dirty_flags & VKD3D_DYNAMIC_STATE_SCISSOR)
+        {
+            VK_CALL(vkCmdSetScissor(list->vk_command_buffer,
+                    0, dyn_state->viewport_count, dyn_state->scissors));
+        }
     }
+    else
+        FIXME_ONCE("Viewport count is zero. Is this allowed?\n");
```
Comment (doitsujin) [inline libs/vkd3d/command.c:4792]: D3D11 allows this; it behaves as if rasterization is disabled so that geometry-only pipelines can still run without viewports. DXVK's not-so-sophisticated way of dealing with this is to bind a dummy viewport with an empty scissor rect; we should probably do the same.

Comment (HansKristian-Work) [reply libs/vkd3d/command.c:4792]: I'll do that then, and split it out to a separate PR.

Comment (HansKristian-Work) [reply libs/vkd3d/command.c]: Fixed.

**Comment by HansKristian-Work** [comment]
Tested on NV as well. Passes fine.

**Review by doitsujin — APPROVED** [review]
